js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx

```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
+                />
+              )}
+            </EvaluationRow>
+          );
+        })}
+      </Box>
+      {(queryResult.data?.autoMaterializeAssetEvaluations?.length || 0) > 0 ? (
+        <PaginationWrapper>
+          <CursorPaginationControls {...paginationProps} />
+        </PaginationWrapper>
+      ) : null}
     </Box>
   );
-};
+}
+
+const RightPanel = ({
+  assetKey,
+  setMaxMaterializationsPerMinute,
+}: {
+  assetKey: Omit<AssetKey, '__typename'>;
+  setMaxMaterializationsPerMinute: (max: number) => void;
+}) => {
+  const queryResult = useQuery<GetPolicyInfoQuery, GetPolicyInfoQueryVariables>(
+    GET_POLICY_INFO_QUERY,
+    {
+      variables: {
+        assetKey,
+      },
+    },
+  );
+  useQueryRefreshAtInterval(queryResult, FIFTEEN_SECONDS);
+  const {data, error} = queryResult;
+
+  React.useEffect(() => {
+    if (data?.assetNodeOrError.__typename === 'AssetNode') {
+      const max = data.assetNodeOrError.autoMaterializePolicy?.maxMaterializationsPerMinute;
+      if (max !== undefined && max !== null) {
+        setMaxMaterializationsPerMinute(max);
+      }
+    }
+  }, [data, setMaxMaterializationsPerMinute]);
 
-const RightPanel = () => {
   return (
-    <Box flex={{direction: 'column'}} style={{maxWidth: '294px'}}>
+    <Box flex={{direction: 'column'}} style={{width: '294px'}}>
       <Box
         padding={{vertical: 16, horizontal: 24}}
         border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}
       >
         <Subheading>Overview</Subheading>
       </Box>
-      <RightPanelSection title="Auto-materialize Policy">
-        <RightPanelDetail
-          title="On missing"
-          value={<Green700Text>True</Green700Text>}
-          tooltip="test"
-        />
-        <RightPanelDetail
-          title="On upstream data"
-          value={<Green700Text>True</Green700Text>}
-          tooltip="test"
-        />
-        <RightPanelDetail
-          title="For freshness"
-          value={<Green700Text>True</Green700Text>}
-          tooltip="test"
-        />
-        <RightPanelDetail title="Maximum materializations per hour" value={2} tooltip="test" />
-      </RightPanelSection>
-      <RightPanelSection title="Freshness Policy">
-        <RightPanelDetail title="Maximum lag minutes" tooltip="test" value={2} />
-        <Box flex={{direction: 'column', gap: 8}}>
-          This asset will be considered late if it is not materialized within 2 minutes of it’s
-          upstream dependencies.
-          <ButtonLink>View upstream assets</ButtonLink>
+      {error ? (
+        <Box padding={24}>
+          <ErrorWrapper>{JSON.stringify(error)}</ErrorWrapper>
+        </Box>
+      ) : !data ? (
+        <Box flex={{direction: 'row', justifyContent: 'center'}} padding={{vertical: 24}}>
+          <Spinner purpose="section" />
         </Box>
-      </RightPanelSection>
+      ) : data.assetNodeOrError.__typename === 'AssetNotFoundError' ? (
+        <Redirect to="/assets" />
+      ) : (
```

Comment:
Worth showing an error message instead of a redirect?

Reply:
hmm, I suppose so